### ISSUE TITLE
New `port` config in `retype.json`

### ISSUE THREAD

**Issue by geoffreymcgill** [retypeapp/retype#8]
Retype needs to allow the custom configuration of the `port` used by the internal webserver during [`retype watch`](https://retype.com/guides/getting_started/#retype-watch) or [`retype run`](https://retype.com/guides/getting_started/#retype-run).

Default is `5000`. 

```json
{
    "port": 5001
}
```

**Comment by geoffreymcgill** [comment]
The new `port` functionality is complete and will be included in the upcoming Retype `v1.5.0` release. 

Two custom port scenarios are supported:

1. Setting the `port` config in your projects `retype.json` file. 
```json
{
    "port": 5001
}
```
2. Passing the `--port 5001` argument in the `retype watch` or `retype run` commands. 
```sh
retype watch --port 5001
```

The default port is still `5000`.

**Comment by dyoder** [comment]
Can we also have an option for it to just find an available port? If the \[port is omitted or set to 0\](https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback), you will get the first available port, which you can then get from the returned server object.

**Comment by geoffreymcgill** [comment]
@dyoder We included similar functionality in `v1.5`. 

The current logic is:

1. Use port 5000 by default. Port `5000` is used because this is a dev tool, and the internal Retype web server is not meant for production. 
2. If port 5000 is in use, increment port value by 1
3. Continue incrementing until an open port is available
4. If `port` is explicitly set in the `retype.json` or explicitly passed as a `--license` command arg, use that port value
5. If the explicitly set port is being used, fail (with friendly error and exit)

Hope this helps.
